testutil/problem_stub.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// ProblemStub describes a subset of a Problem, and is used to verify
+// that problems look correct in tests.
+type ProblemStub struct {
+	Descriptor desc.Descriptor
+	Message    string
+}
+
+// VerifyDescriptor verifies that the stub's descriptor and
+// problem's descriptor match.
+func (ps *ProblemStub) VerifyDescriptor(p lint.Problem, t *testing.T) {
+	if !reflect.DeepEqual(p.Descriptor, ps.Descriptor) {
+		t.Errorf("Expected descriptor %v; got %v", ps.Descriptor, p.Descriptor)
+	}
+}
+
+// VerifyMessage verifies that the stub's message is included in the
+// problem's message.
+func (ps *ProblemStub) VerifyMessage(p lint.Problem, t *testing.T) {
```

Comment:
It looks like [change detectors](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html). I don't think we want to verify the *exact* message, maybe just to check if the message contains some required information. For example, we more likely want to see if the problem returns the right suggestion.

Reply:
Completely agree, I oppose change detector tests. If you notice, it checks for _containment_, not _equivalence_. And if you look at the test for AIP-131, you will notice that I am testing for a single word -- what the message says the right value is.